### ISSUE TITLE
Implement error_tracking API

### ISSUE THREAD

**Issue by svenstaro** [gitlabhq/terraform-provider-gitlab#457]
There is currently no support for managing the Sentry integration via the `error_tracking` API https://docs.gitlab.com/ee/api/error_tracking.html. Seems pretty easy to add, though.

**Comment by rescribet** [comment]
Would be useful to be able to read the gitlab DSN to set-up error reporting automatically.

**Comment by Shocktrooper** [comment]
@timofurrer Am I correct in assuming we would need upstream work for this in [go-gitlab](https://github.com/xanzy/go-gitlab)? Probably a new connector as I don't think this would fit into any of the others

**Comment by timofurrer** [comment]
@Shocktrooper Yes, if it's not already in go-gitlab - it needs to support it first. We've decided to exclusively use go-gitlab :)

**Comment by Shocktrooper** [comment]
@timofurrer / @armsnyder / @PatrickRice-KSC  How do we want to test this.  I was playing around with the go-gitlab library directly as I was developing it and it seems unless you turn on the [feature flag for integrated backends](https://docs.gitlab.com/ee/operations/error_tracking.html#:~:text=By%20default%20this%20feature%20is%20not%20available.%20To%20make%20it%20available%20on%20self%2Dmanaged%20GitLab%2C%20ask%20an%20administrator%20to%20enable%20the%20feature%20flag%20named%20integrated_error_tracking.%20The%20feature%20is%20not%20ready%20for%20production%20use.%20On%20GitLab.com%2C%20this%20feature%20is%20not%20available.) you cannot turn-on/configure the integation at all either via the UI or the API without having a live sentry instance that you can get configuration values from.  Attempting to enable or configure the integration without sentry and without the integrated backend results in a 404 

`2022/05/19 18:40:22 PATCH https://somegitlabinstance.com/api/v4/projects/123/error_tracking/settings: 404 {message: 404 Error Tracking Setting Not Found}`

Going forward should we just turn on the feature flag in the startup script and just test this without sentry and only use the integrated backend? What are your thoughts?

**Comment by timofurrer** [comment]
> Going forward we should just turn on the feature flag in the startup script and just test this without sentry and only use the integrated backend.

Yes, I agree. We should avoid having any external services required for our automated testing!

I've just reviewed the PR in go-gitlab, too ;)
